bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/commands/StartEditLocalDocumentHandler.java

```diff
@@ -135,14 +137,46 @@ private boolean tryHandleExternalIfApplicable(Object lockObject) {
 			} else if (lockObject instanceof IDocumentLetter) {
 				document = (IDocumentLetter) lockObject;
 			} else {
-				LoggerFactory.getLogger(getClass()).error("Invalid argument [{}]", lockObject.getClass());
+				LoggerFactory.getLogger(getClass()).error("Invalid argument [{}]", lockObject.getClass()); //$NON-NLS-1$
 				return false;
 			}
 
 			IVirtualFilesystemHandle handle = DocumentLetterUtil.getExternalHandleIfApplicable(document);
 			Optional<File> file = handle.toFile();
 			if (file.isPresent()) {
-				Program.launch(file.get().getAbsolutePath());
+				String extension = handle.getExtension();
+				// Check whether we want to convert the generated docx file directly to PDF and
+				// txt when running GUI-Tests
+				String convert2pdf = System.getProperty("ch.elexis.convertDocx2PDF", StringUtils.EMPTY); //$NON-NLS-1$
+				if (extension.equalsIgnoreCase("docx") && //$NON-NLS-1$
```

Comment:
Bitte Methode isConvertDocx2Pdf erstellen und da die Prüfung des sys property und der file extension implementierten.

Dann den Impl. der Konvertierung in eine convertDocx2Pdf Methode verschieben.

Dadurch bleibt die Methode hier übersichtlicher.